components/Datepicker/src/index.tsx

```diff
@@ -0,0 +1,188 @@
+   */
+  disabled?: boolean;
+  /**
+   * Locale (from @gemeente-denhaag/datepicker/locale or date-fns/locale) (default: nl)
+   */
+  locale?: Locale;
+  /**
+   * Starting date for calendar (default: today)
+   */
+  startDate?: Date;
+}
+
+interface DatepickerState {
+  current: Date;
+  selected?: Date;
+  opened: boolean;
+}
+
+/**
+ * The DatePicker component allows the user to select a date.
+ */
+export const Datepicker: React.FC<DatepickerProps> = ({
+  startDate = new Date(),
+  locale = nl,
+  ...props
+}: DatepickerProps) => {
+  const rootStyles = clsx(
+    'denhaag-datepicker',
+    {
+      'denhaag-datepicker--error': props.error,
+      'denhaag-datepicker--disabled': props.disabled,
+    },
+    props.className,
+  );
+
+  const [state, setState] = React.useState<DatepickerState>({
+    current: setDate(startDate, 1),
+    selected: undefined,
+    opened: false,
+  });
+
+  const componentRef = React.useRef<HTMLDivElement>(null);
+
+  const outsideClickListener = (e: Event) => {
+    if (!componentRef.current?.contains(e.target as Node)) {
+      setState({
+        ...state,
+        opened: false,
+      });
+    }
+  };
+
+  React.useEffect(() => {
+    document.addEventListener('click', outsideClickListener);
+    return () => {
+      document.removeEventListener('click', outsideClickListener);
+    };
+  }, [componentRef]);
+
+  const renderDay = (i: number, j: number) => {
+    const date = setDay(addWeeks(state.current, i), (j + (locale.options?.weekStartsOn || 0)) % 7, { locale: locale });
+    if (isSameMonth(date, state.current)) {
+      return (
+        <button
+          type="button"
+          onClick={() => {
+            setState({ ...state, selected: date, opened: false });
+          }}
+        >
+          {date.getDate()}
+        </button>
+      );
+    }
+    return '';
+  };
+
+  return (
+    <div className={rootStyles} ref={componentRef}>
+      <input
+        disabled={props.disabled}
+        type="date"
+        className="denhaag-datepicker__input"
+        value={state.selected ? formatISO(state.selected, { representation: 'date' }) : ''}
+        onChange={(event) => {
+          setState({
+            ...state,
+            current:
+              event.target.value && new Date(event.target.value).getDate()
+                ? setDate(new Date(event.target.value), 1)
+                : state.current,
+            selected:
+              event.target.value !== undefined
+                ? new Date(event.target.value).getDate()
+                  ? new Date(event.target.value)
+                  : state.selected
+                : undefined,
+          });
+        }}
+        onClick={(event) => {
+          event.preventDefault();
+          setState({ ...state, opened: true });
+        }}
+        onKeyDown={(event) => {
+          if (event.key === 'Enter') {
+            event.preventDefault();
+            setState({ ...state, opened: true });
+          }
+        }}
```

Comment:
Pressing `Space` on Chrome (Win 11) opens the browser datepicker. 
![image](https://user-images.githubusercontent.com/24610692/145600278-35747a56-c3d3-4203-9c8d-b35dbf98cda5.png)

Reply:
Fixed with the implementation of all keyboard functionality.